### PR TITLE
(Merge 3/9 in afternoon) Feat: go agent 3-20-4 release

### PR DESCRIPTION
**Pull request by iamemilio**
Feature release for go agent 3.20.4 release

### CODE REVIEW

**Comment by ally-sassman** [comment]
@iamemilio Can you confirm when these release notes should be merged?

**Comment by iamemilio** [comment]
EOD today would be best. Thanks!

File: src/content/docs/apm/agents/go-agent/configuration/go-agent-configuration.mdx [hunk 1]
```diff
@@ -1591,6 +1591,79 @@ The following settings are used to configure the error collector:
...
+          < td>
+            `newrelic.Config` struct
+          </td>
+        </tr>
+      </tbody>
+    </table>
+
+    This controls which HTTP response codes are expected as errors.
+
+    Response codes that are expected will not affect your applications apdex or error alerts, but will still be recorded.
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/go-agent/configuration/go-agent-configuration.mdx:1634]:
```suggestion
    Response codes that are expected will not affect your application's apdex or error alerts, but they will still be recorded.
```

File: src/content/docs/apm/agents/go-agent/configuration/go-agent-configuration.mdx [hunk 2]
```diff
@@ -1591,6 +1591,79 @@ The following settings are used to configure the error collector:
...
+    This function's default form is:
+
+    ```go
+    config.ErrorCollector.ExpectStatusCodes = []int{
+        100,
+        http.StatusAccepted,
+    }
+    ```
+
+    You can also add response codes as HTTPs, as `http.StatusAccepted` above.
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/go-agent/configuration/go-agent-configuration.mdx:1645]:
```suggestion
    You can also add response codes as HTTPs such as `http.StatusAccepted` described above.
```

File: src/content/docs/apm/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected.mdx [hunk 3]
```diff
@@ -38,7 +38,7 @@ To learn how to get an APM agent to report an error, see the agent-specific API
 Sometimes the APM agent instruments an error that you don't want reported, such as errors that contain sensitive information like user login errors. If you don't want an error to report to our [collector](/docs/accounts-partnerships/education/getting-started-new-relic/glossary#collector), you can ignore the error, and the APM agent discards the error entirely.
 
 <Callout variant="tip">
-  For Java, Ruby, Node.js, and Python: If you want to report errors to APM but don't want those errors to affect your Apdex or error rate, [mark them as expected](#expected-links) instead.
+  For Java, Ruby, Node.js, Go and Python: If you want to report errors to APM but don't want those errors to affect your Apdex or error rate, [mark them as expected](#expected-links) instead.
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected.mdx:41]:
```suggestion
  For Java, Ruby, Node.js, Go, and Python: If you want to report errors to APM but don't want those errors to affect your Apdex or error rate, [mark them as expected](#expected-links) instead.
```

File: src/content/docs/apm/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected.mdx [hunk 4]
```diff
@@ -82,6 +82,7 @@ To configure errors as expected, see the agent-specific documentation:
 * [Ruby](/docs/agents/ruby-agent/configuration/ruby-agent-configuration/#error-collector)
 * [Node.js](/docs/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration)
 * [.NET](/docs/agents/net-agent/configuration/net-agent-configuration#error_collector)
+* [Go](/docs/agents/go-agent/configuration/go-agent-configuration#error_collector)
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected.mdx:85]:
```suggestion
* [Go](/docs/apm/agents/go-agent/configuration/go-agent-configuration/#error-collector)
```

File: src/content/docs/apm/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected.mdx [hunk 5]
```diff
@@ -82,6 +82,7 @@ To configure errors as expected, see the agent-specific documentation:
 * [Ruby](/docs/agents/ruby-agent/configuration/ruby-agent-configuration/#error-collector)
 * [Node.js](/docs/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration)
 * [.NET](/docs/agents/net-agent/configuration/net-agent-configuration#error_collector)
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected.mdx:84]:
```suggestion
* [.NET](/docs/apm/agents/net-agent/configuration/net-agent-configuration/#error_collector)
```

File: src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+releaseDate: '2023-03-09'
+version: 3.20.4
+downloadLink: 'https://github.com/newrelic/go-agent/tree/v3.20.4'
+features: ["Improved ignore error performence", "Expected HTTP status code errors"]
+bugs: ["Error message truncation", "Span event limit config option"]
+security: ["aws sdk go unencrypted hash vulnerability"]
+---
+
+<Callout variant="important">
+We recommend updating to the latest agent version as soon as it's available. If your organization has established practices that prevent you from upgrading to the latest version, ensure that your agents are regularly updated to a version at most 90 days old. Read more about [keeping your agent up to date](/docs/new-relic-solutions/new-relic-one/install-configure/update-new-relic-agent/).
```
Comment (ally-sassman) [inline src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx:12]:
```suggestion
We recommend updating to the latest agent version as soon as it's available. If your organization has established practices that prevent you from upgrading to the latest version, ensure that your agents are regularly updated to a version that's at most 90 days old. Read more about [keeping your agent up to date](/docs/new-relic-solutions/new-relic-one/install-configure/update-new-relic-agent/).
```

File: src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+</Callout>
+
+### Fixed
+* nrmssql driver updated to use version maintained by Microsoft
+* bug where error messages were not truncated to the maximum size, and would get dropped if they were too large
+* bug #643 where number of span events was hard coded to 1000, and config setting was being ignored
+* unencrypted hash vulnerability in v1.15.0 of github.com/aws/aws-sdk-go patched by bumping to v1.33.0
+
+### Added
+* improved performance of ignore error code checks in agent
```
Comment (ally-sassman) [inline src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx:28]:
```suggestion
* Improved performance of ignore error code checks in agent
```

File: src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+<Callout variant="important">
+We recommend updating to the latest agent version as soon as it's available. If your organization has established practices that prevent you from upgrading to the latest version, ensure that your agents are regularly updated to a version at most 90 days old. Read more about [keeping your agent up to date](/docs/new-relic-solutions/new-relic-one/install-configure/update-new-relic-agent/).
+</Callout>
+
+## 3.20.4
+
+<Callout variant="important">
+ Please note that the v2 go agent is no longer supported according to our EOL policy.
```
Comment (ally-sassman) [inline src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx:18]:
```suggestion
 Please note that the v2 Go agent is no longer supported according to our EOL policy. 
```

File: src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+<Callout variant="important">
+ Please note that the v2 go agent is no longer supported according to our EOL policy. 
+</Callout>
+
+### Fixed
+* nrmssql driver updated to use version maintained by Microsoft
+* bug where error messages were not truncated to the maximum size, and would get dropped if they were too large
+* bug #643 where number of span events was hard coded to 1000, and config setting was being ignored
+* unencrypted hash vulnerability in v1.15.0 of github.com/aws/aws-sdk-go patched by bumping to v1.33.0
```
Comment (ally-sassman) [inline src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-20-4.mdx:25]:
```suggestion
* nrmssql driver updated to use version maintained by Microsoft
* Bug where error messages were not truncated to the maximum size, and would get dropped if they were too large
* Bug #643 where number of span events was hard coded to 1000, and config setting was being ignored
* Unencrypted hash vulnerability in v1.15.0 of github.com/aws/aws-sdk-go patched by bumping to v1.33.0
```

**Review by ally-sassman — APPROVED** [review]

**Pull request merged**